src/postamble.js

```diff
@@ -316,6 +316,10 @@ Module['exit'] = Module.exit = exit;
 var abortDecorators = [];
 
 function abort(what) {
+  if (Module['onRuntimeInitializeFailed']) {
```

Comment:
I wonder though if we shouldn't add `onAbort` instead of this. Would it handle your use case? If so, then since it handles more use cases too, and avoids the complexity of needing to unset `onRuntimeInitializeFailed`, it seems superior.